src/menu/actions.c

```diff
@@ -24,8 +24,21 @@ static void actions_clear (menu_t *menu) {
 }
 
 static void actions_update_direction (menu_t *menu) {
-    joypad_8way_t held_dir = joypad_get_direction(JOYPAD_PORT_1, JOYPAD_2D_DPAD | JOYPAD_2D_STICK);
-    joypad_8way_t fast_dir = joypad_get_direction(JOYPAD_PORT_1, JOYPAD_2D_C);
+    int prev_pad = -1;
+    joypad_8way_t held_dir = JOYPAD_8WAY_NONE;
+    joypad_8way_t fast_dir = JOYPAD_8WAY_NONE;
+
+    JOYPAD_PORT_FOREACH (i) {
+        joypad_8way_t held_dir_temp = joypad_get_direction(i, JOYPAD_2D_DPAD | JOYPAD_2D_STICK);
+        joypad_8way_t fast_dir_temp = joypad_get_direction(i, JOYPAD_2D_C);
+        if ((held_dir_temp != JOYPAD_8WAY_NONE || prev_pad == i) && held_dir_temp != held_dir) {
```

Comment:
Same thing as below. Not sure what `prev_pad == i` accomplishes because from my knowledge this check will never return true.